File: leetcode/easy/cpp/LC_2423.cpp
#include <iostream>
#include <unordered_map>
#include <unordered_set>
#include <vector>

using namespace std;

class Solution {
public:
    bool equalFrequency(string word) {
        unordered_map<char, int> counter;
        for(char i: word) {
            counter[i]++;
        }
        vector<int> frequencies;

        for(auto& p: counter) {
            frequencies.push_back(p.second);
        }

        for(int i=0; i<frequencies.size(); i++) {
            frequencies[i]--;
            
            unordered_set<int> freqSet;
            for(int freq: frequencies) {
                if(freq > 0) {
                    freqSet.insert(freq);
                }
            }

            if(freqSet.size() == 1) {
                return true;
            }

            frequencies[i]++;
        }

        return false;

    }
};

int main() {
    Solution solution;
    string word;
    cin >> word;
    cout << (solution.equalFrequency(word) ? "true" : "false") << endl;
    return 0;
}

File: leetcode/easy/python/LC_2423.py
from collections import defaultdict

class Solution:
    def equalFrequency(self, word: str) -> bool:
        counter = defaultdict(int)
        for i in word:
            counter[i] += 1
        
        # counter = {"a": 1, "b": 1, "c": 2}
        frequencies = list(counter.values())
        
        for i in range(len(frequencies)):
            frequencies[i] -= 1
            if len(set(frequencies)) == 1:
                return True
            # Case: were counter = {"a": 1, "b": 2, "c": 2}
            if frequencies[i] == 0 and len(set(frequencies)) == 2:
                return True
            frequencies[i] += 1
        
        return False

solution = Solution()
word = input()
print(solution.equalFrequency(word))